lib/rspec/buildkite/analytics/session.rb

```diff
@@ -12,6 +12,28 @@ class Session
     class RejectedSubscription < StandardError; end
     class InitialConnectionFailure < StandardError; end
 
+    class Logger
+      def initialize
+        @log = Queue.new
+      end
+
+      def write(str)
+        @log << "#{Time.now.strftime("%F-%R:%S.%9N")} #{Thread.current} #{str}"
+      end
+
+      def to_array
+        # This empty check is important cos calling pop on a Queue is blocking until
+        # it's not empty
+        if @log.empty?
+          []
+        else
+          Array.new(@log.size) { @log.pop }
```

Comment:
Would this only pops the last element from the queue instead of all elements from the queue? 🤔

Reply:
If you pass a block to `Array.new` it populates it with the given object, and since I set the size of the Array to be the size of the queue, pop will be called the right number of times. There's further examples at the top of [this page](https://ruby-doc.org/core-2.7.0/Array.html)

I found this code snippet on [this stackoverflow post](https://stackoverflow.com/questions/34481288/ruby-queue-to-array), happy to choose a different way of doing it if you feel like it's unclear?

Reply:
Ah what I meant is suppose we have 2 errors in `@log`:

```ruby
@log << "error 1"
@log << "error 2"
```

This line will create any array of `["error 1", "error 1"]` instead of `["error 1", "error 2"]` right?

Is this what we want? 🤔

Reply:
Ruby `Queue` is a FIFO queue and the pop method removes the first item in the queue, [docs here](https://ruby-doc.org/core-2.5.0/Queue.html#method-i-pop). Also I attached a log file so you can see that it's working :)

Reply:
The log file looks good! 👍🏼